source/transfer_list.rst

```diff
@@ -714,3 +715,56 @@ semantics should be consistent (but not limited to) with [UEFI]_ and [ACPI]_.
         } GuidBlob;
 
 The size of the BLOB content (`n`) is `data_size - 16`.
+
+
+.. _guid_pointer_entry:
+
+GUID POINTER entry layout (XFERLIST_GUID_POINTER)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type points to an object identified by its GUID. The values and
+semantics should be consistent (but not limited to) with [UEFI]_ and [ACPI]_.
+
+.. _tab_guid_pointer:
+.. list-table:: GUID POINTER type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **6**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of `GuidPointer` in bytes.
+
+   * - guid_blob
+     - 0x20
+     - hdr_size
+     - The guid_blob field contains a `GuidPointer`.
+
+.. code-block:: c
+
+        typedef struct {
+          uint32_t data1;
+          uint16_t data2;
+          uint16_t data3;
+          uint8_t data4[8];
+        } guid;
+
+        typedef struct {
+          guid name;          // GUID name
+          uint64_t address;   // content address
+          uint64_t size;      // content size
```

Comment:
What is the goal of putting this in the transfer list? We are trying to make the data visible and documented

Reply:
I put a more detailed rationale in the description. The pointer variant would be useful if you want to keep an object at a specific location in memory. It's also an opportunity for optimizations: avoid moving content memory, avoid `XFERLIST_VOID` some entries due to alignment, etc.

Reply:
I am not sure I understood the part about "visible and documented" w.r.t. this part of the PR.

Reply:
Let me know if the updates did not address your question.